crates/cheatcodes/src/inspector.rs

```diff
@@ -1361,7 +1359,17 @@ impl Cheatcodes {
 
         // Remove recorded gas if we exit frame.
         if will_exit(interpreter.instruction_result) {
-            self.paused_frame_gas.pop();
+            if self.pause_gas_metering {
+                self.paused_frame_gas.pop();
+            } else {
```

Comment:
the point of this fn is that it's outlined with `#[cold]`, I'd add a separate bool for handling this branch because it can only happen if pauseGasMetering has been called in the past that remains true

Reply:
kk, will add a resumed_gas flag to apply such

Reply:
ptal https://github.com/foundry-rs/foundry/pull/8744/commits I externalized gas metering sate in `GasMetering ` struct for preparing to support https://github.com/foundry-rs/foundry/issues/4211 (would make sense to move `last_call_gas` in there too)